### PR TITLE
Remove g:signify_sign_show_text

### PR DESCRIPTION
**Pull request by mhinz**
Using Unicode block characters like █ is much easier than using
`g:signify_sign_show_text = 0` and creating extra highlight groups for it.

Closes https://github.com/mhinz/vim-signify/issues/374

### CODE REVIEW

**Comment by mhinz** [comment]
@jamessan

Would you be opposed to making this the new default:

```
  let g:signify_sign_add = '▊'
  let g:signify_sign_change = '▊'
  let g:signify_sign_delete = '▊'
```

I think that would give the plugin a more modern look. I'm sure there will be someone who prefers the "old " ASCII chars or people don't want to distinguish the signs based on colors alone, but that's what these options are for anyway.

I think (read: hope) the majority of users would like this. Choosing defaults is hard, though.

**Comment by jamessan** [comment]
TL;DR: I'm probably -0 on that and find it amusing that changing `g:signify_show_text`'s default (and fixing the definition of the signs) would have achieved the same, while make it easy to flip back to the other behavior.

> I think that would give the plugin a more modern look.

Not really sure what that means. :)

> Would you be opposed to making this the new default:

In other words, you want `g:signify_sign_show_text = 0` to be default? ;)

> I'm sure there will be someone who prefers the "old " ASCII chars or people don't want to distinguish the signs based on colors alone

Or _can't_ distinguish by color alone.  Text/shape is more accessible than color.

That's one of the things that annoys me about GitHub's icons for PR status in the notifications list.  The "merged PR" is a different shape than the rest, so that's easy to distinguish.  However, the only difference between the "open PR" and "closed PR" icons is that one is green and the other red.  There's not enough "body" to the icons (unlike the issue icon) for me to easily tell the difference.

Granted, the lack of "body" isn't an issue for the characters you're choosing, but the shift to only color may be.  I suppose people who may be affected by that would already have a colorscheme that's adapted to their needs.

**Comment by mhinz** [comment]
> [...] find it amusing that changing `g:signify_show_text`'s default (and fixing the definition of the signs) would have achieved the same, while make it easy to flip back to the other behavior.

Well, removing an (not very popular) option was another reason. Especially when achieving the same effect using other options is arguably easier.

Now you just need to set options for using different signs vs. using `g:signify_show_text = 0` + using extra highlight groups (because colorschemes just set different fg colors and usually use the same bg color for all signs, the same as for `SignColumn`). And many people don't like to fiddle with colors.

> Or can't distinguish by color alone.

Yeah, that's actually what I thought but failed to put in words.

> Text/shape is more accessible than color.

I agree (and was 99% sure you wouldn't be in favor of such defaults anyway. :-)).

> That's one of the things that annoys me about GitHub's icons for PR status in the notifications list. The "merged PR" is a different shape than the rest, so that's easy to distinguish. However, the only difference between the "open PR" and "closed PR" icons is that one is green and the other red. There's not enough "body" to the icons (unlike the issue icon) for me to easily tell the difference in the default light theme. The dark themes show the contrast much better.

Hmm, I never paid much attention to that before, but considering that dyschromatopsia (I hope that's the right word for the problem of distinguishing between red and green) is among the more common visual impairments, I can see how using red and green for the same icon is a huge issue indeed.

No change in sign defaults!

**Comment by jamessan** [comment]
Yay, GitHub finally [fixed](https://github.blog/changelog/2021-06-08-new-issue-and-pull-request-state-icons) the icon issue.
